stat/stat.go

```diff
@@ -217,6 +217,59 @@ func Correlation(x, y, weights []float64) float64 {
 	return (sxy - xcompensation*ycompensation/sumWeights) / math.Sqrt(sxx*syy)
 }
 
+// KendallCorrelation returns the weighted Tau-a Kendall correlation between the
+// samples of x and y.
```

Comment:
I think there needs to be some comment that the weight of a pair is equal to weights[i] * weights[j]. This may require an extended explanation of what Kendal is.